From: Mathieu Trudel-Lapierre <mathieu.trudel-lapierre@canonical.com>
Subject: applet: clear notification actions before adding new ones
Bug-Ubuntu: http://bugs.launchpad.net/bugs/606825 

Make sure we only ever have one of the "Don't show this again" button that
will trigger disabling notifications. Otherwise we end up with multiple
instances of the same button in notifications because one gets added every
time a new notification stacks on top of an existing one.

Index: network-manager-applet-0.9.1.90/src/applet.c
===================================================================
--- network-manager-applet-0.9.1.90.orig/src/applet.c	2012-01-06 12:03:08.000000000 +0100
+++ network-manager-applet-0.9.1.90/src/applet.c	2012-01-06 12:03:30.944469038 +0100
@@ -876,6 +876,7 @@
 	notify_notification_set_timeout (notify, NOTIFY_EXPIRES_DEFAULT);
 
 	if (applet_notify_server_has_actions () && action1) {
+		notify_notification_clear_actions (notify);
 		notify_notification_add_action (notify, action1, action1_label,
 		                                action1_cb, action1_user_data, NULL);
 	}
